src/tree/mod.rs

```diff
@@ -87,18 +87,19 @@ where
         let mut handles = vec![];
         handles.push(self.canonical_tree_manager.spawn(leaf_updates_tx));
 
-        //TODO: handle if there are no bridged roots to spawn, otherwise the channel will close
-        for bridged_tree in self.bridged_tree_manager.iter() {
-            handles.push(bridged_tree.spawn(bridged_root_tx.clone()));
+        if !self.bridged_tree_manager.is_empty() {
```

Comment:
This isn't actually any different right? If `self.bridged_tree_manager` is empty then iterating over it won't yield any items so the loop never executes?

Reply:
Yes, you are right, if it is empty it will not yield any items, however in the previous implementation, we would still spawn `handle_bridged_updates` which would cause issues since the `Sender` will be dropped. 

Technically, the only thing that needs to be within the `!self.bridged_tree_manager.is_empty(){}`block is spawning the task to handle updates, but I put everything to make it more cohesive.